### ISSUE TITLE
[🔥] PHONE AUTH - Cannot create PhoneAuthCredential without either verificationProof, sessionInfo, temporary proof, or enrollment ID 

### ISSUE THREAD

**Issue by Monir-Shembesh** [invertase/react-native-firebase#5279]
## Issue

I have been trying to `confirm` my verification code that I receive after `signInWithPhoneNumber` in auth but I get: 

`[auth/unknown] Cannot create PhoneAuthCredential without either verificationProof, sessionInfo, temporary proof, or enrollment ID.`

I have tried multiple ways and different methods but nothing worked and I have tried it on the latest packages release as well as older ones such as `v10.3.0`

This is the code I have used. Keep in mind that I receive my phone auth code but wont verify even with test codes: 

 ```
const sendOtp = async () => {
    try {
      const sendOtpResults = await auth().signInWithPhoneNumber(phoneNumber);
      setConfirm(sendOtpResults);
      console.log(JSON.stringify(sendOtpResults));
    } catch (error) {
      console.log(JSON.stringify(error));
    }
  };

  const confirmPhoneNumber = async (code: any) => {
    const response = await confirm.confirm(code);
    console.log(JSON.stringify(response));
  };
```

---

### Javascript

<details><summary>Click To Expand</summary>
<p>

#### `package.json`:

```json
# N/A
```

#### `firebase.json` for react-native-firebase v6:

```json
# N/A
```

</details>

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
# N/A
```

#### `AppDelegate.m`:

```objc
// N/A
```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

- [ x ] my application is an AndroidX application?
- [ x ] I am using `android/gradle.settings` `jetifier=true` for Android compatibility?
- [ x ] I am using the NPM package `jetifier` for react-native compatibility?

#### `android/build.gradle`:

```groovy
// N/A
```

#### `android/app/build.gradle`:

```groovy
// N/A
```

#### `android/settings.gradle`:

```groovy
// N/A
```

#### `MainApplication.java`:

```java
// N/A
```

#### `AndroidManifest.xml`:

```xml
<!-- N/A -->
```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
 System:
    OS: Windows 10 10.0.19042
    CPU: (8) x64 Intel(R) Core(TM) i7-4790K CPU @ 4.00GHz
    Memory: 6.27 GB / 15.95 GB
  Binaries:
    Node: 14.16.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 6.14.11 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK: Not Found
  IDEs:
    Android Studio: Not Found
    Visual Studio: Not Found
  Languages:
    Java: 15.0.2 - /c/Program Files/Common Files/Oracle/Java/javapath/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1
    react-native: 0.64.0 => 0.64.0
    react-native-windows: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

```

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [ ] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ x ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - `e.g. 10..3.0`
- **`Firebase` module(s) you're using that has the issue:**
  - `auth`
- **Are you using `TypeScript`?**
  - N`

</p>
</details>

---

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

**Comment by mikehardy** [comment]
Hi there - this method is tested every time we make a change anywhere in the codebase, and works:

https://github.com/invertase/react-native-firebase/blob/221eb5001f69fb3c8586e4cdef4e74dad3380c08/packages/auth/e2e/phone.e2e.js#L58-L64

I'm not sure what's going on with that code or in your project but I suspect it's project-specific 🤔

**Comment by Monir-Shembesh** [comment]
Hi @mikehardy yes It is weird. Does my code look correct to you? also I am using react native 0.64 could that be the issue

**Comment by mikehardy** [comment]
Our tests app may be used to verify at all times, it is using 0.64 (actually, 0.64.1 already as I'm on Xcode 12.5 for iOS and I had to bump it ASAP to get the build working locally). 

https://github.com/invertase/react-native-firebase/blob/221eb5001f69fb3c8586e4cdef4e74dad3380c08/tests/package.json#L28

I use phone verification (not sign in, but it's the same) in my app and it works. I do consulting for another app that does phone verification and it's on react-native-firebase 11.4.1 + react-native 0.64.1 and it's working fine as well, including passing manual QA in an overnight cycle last night.

So I'm never 100% about software (it always surprises!) but I'm pretty sure there's something wrong with your project or code. Your code is not a minimal reproducible example (https://stackoverflow.com/help/minimal-reproducible-example) and you skipped the template so I won't speculate. But I've posted code that works, so hopefully that helps

**Comment by Monir-Shembesh** [comment]
@mikehardy yh it is weird too and when I use phone verification the app just keeps resending OTP codes every like 2 minutes

**Comment by mikehardy** [comment]
That sounds terribly wrong. Your code definitely has issues. I'm going to close this since it lacks issue template and there's no reproduction. If you can post a self-contained App.js we can drop in a project to reproduce a problem, we may reopen and investigate

**Comment by Monir-Shembesh** [comment]
Hi @mikehardy I am here to report that it is working fine. All I did was using changing my component to class instead of a functional component and the same code ran and worked fine. Very weird but ok 1 class component wouldn't hurt during registration I guess. One more question if you dont mind, in IOS with silent notification I dont get recaptchahs at all but with android why do I get recaptcahs all the time? I have configured my google cloud with `safetyNet` but still most of the times i get recaptchas.
